# Task 1.3 (Statistical Analysis using Python)

## Batch C Group 1
### Anjum Shagufta Azad

## Q1 :Descriptive statistics for both numerical and categorical and draw few insights from them.

In [ ]:
#import linear algebra and data manipulation libraries
import numpy as np
import pandas as pd

#import standard visualization
import matplotlib.pyplot as plt
import seaborn as sns

#import statistical analysis libraries
import scipy.stats as stats
import math

In [ ]:
df1 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\City_Master.csv")

In [ ]:
df1.head()

In [ ]:
df2 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Customer_and_bank details_p1.csv")
df3 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Customer_campaign_details_p1.csv")
df4= pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Customer_Postal_code_details.csv")
df5 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Customer_Response_data_p1.csv")
df6 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Customer_social_economic_data_p1.csv")
df7 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\Region_code_master.csv")
df8 = pd.read_csv("C:\\Users\\shazad\\Downloads\\Capstone Project1_Marketing Data\\Capstone Project1_Marketing Data\\Marketing_Campaign_P1\\State_Master.csv")

In [ ]:
merged = pd.merge(df1, df2)
merge1 = pd.merge(df2,df3)
merge2 = pd.merge(merge1, df5)
merge3 = pd.merge(merge2, df6)

#### Merge files: Customer & Bank Details,Customer-Campaign-Details, Customer_Social_Economic_Data, Response Data

In [16]:
merge3

Customer_id  age          job  marital            education  default  \
0                1   56     services  married          high.school       no   
1                2   45     services  married             basic.9y  unknown   
2                3   59       admin.  married  professional.course       no   
3                4   41  blue-collar  married              unknown  unknown   
4                5   24   technician   single  professional.course       no   
...            ...  ...          ...      ...                  ...      ...   
37079        37080   73      retired  married  professional.course       no   
37080        37081   46  blue-collar  married  professional.course       no   
37081        37082   56      retired  married    university.degree       no   
37082        37083   44   technician  married  professional.course       no   
37083        37084   74      retired  married  professional.course       no   

      housing loan Region_Code State_Code  ... campaign pdays previous  \
0          no  yes           3         S1  ...        1   999        0   
1          no   no           3         S1  ...        1   999        0   
2          no   no           4         S2  ...        1   999        0   
3          no   no           3         S3  ...        1   999        0   
4         yes   no           3         S3  ...        1   999        0   
...       ...  ...         ...        ...  ...      ...   ...      ...   
37079     yes   no           2        S16  ...        1   999        0   
37080      no   no           2        S16  ...        1   999        0   
37081     yes   no           2        S16  ...        2   999        0   
37082      no   no           4        S17  ...        1   999        0   
37083     yes   no           1         S6  ...        3   999        1   

          poutcome    y  emp.var.rate  cons.price.idx  cons.conf.idx  \
0      nonexistent   no           1.1          93.994          -36.4   
1      nonexistent   no           1.1          93.994          -36.4   
2      nonexistent   no           1.1          93.994          -36.4   
3      nonexistent   no           1.1          93.994          -36.4   
4      nonexistent   no           1.1          93.994          -36.4   
...            ...  ...           ...             ...            ...   
37079  nonexistent  yes          -1.1          94.767          -50.8   
37080  nonexistent   no          -1.1          94.767          -50.8   
37081  nonexistent   no          -1.1          94.767          -50.8   
37082  nonexistent  yes          -1.1          94.767          -50.8   
37083      failure   no          -1.1          94.767          -50.8   

      euribor3m nr.employed  
0         4.857      5191.0  
1         4.857      5191.0  
2         4.857      5191.0  
3         4.857      5191.0  
4         4.857      5191.0  
...         ...         ...  
37079     1.028      4963.6  
37080     1.028      4963.6  
37081     1.028      4963.6  
37082     1.028      4963.6  
37083     1.028      4963.6  

[37084 rows x 25 columns]

In [17]:
merge3.shape


(37084, 25)

In [18]:
 merge3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 37084 entries, 0 to 37083
Data columns (total 25 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Customer_id     37084 non-null  int64  
 1   age             37084 non-null  int64  
 2   job             37084 non-null  object 
 3   marital         37084 non-null  object 
 4   education       37084 non-null  object 
 5   default         37084 non-null  object 
 6   housing         37084 non-null  object 
 7   loan            37084 non-null  object 
 8   Region_Code     37084 non-null  object 
 9   State_Code      37084 non-null  object 
 10  City_Code       37084 non-null  object 
 11  contact         37084 non-null  object 
 12  month           37084 non-null  object 
 13  day_of_week     37084 non-null  object 
 14  duration        37084 non-null  int64  
 15  campaign        37084 non-null  int64  
 16  pdays           37084 non-null  int64  
 17  previous        37084 non-null 

### Find percentage of missing values for each column

In [19]:

missing_values = merge3.isnull().mean()*100

missing_values.sum()

0.0

In [20]:
merge3['job'].value_counts()

admin.           9420
blue-collar      8314
technician       6096
services         3547
management       2637
retired          1541
entrepreneur     1309
self-employed    1276
housemaid         946
unemployed        925
student           767
unknown           306
Name: job, dtype: int64

### According to the data most of the population is in admin profession and the least population is in student.

In [21]:
merge3['marital'].value_counts()

married     22479
single      10407
divorced     4126
unknown        72
Name: marital, dtype: int64

### Most of the population is married and least are the divorced.

In [22]:
merge3['education'].value_counts()

university.degree      10971
high.school             8542
basic.9y                5421
professional.course     4746
basic.4y                3765
basic.6y                2074
unknown                 1549
illiterate                16
Name: education, dtype: int64

### Most of the population has university degree i.e. 10971 and the illiterate in the population are 16 only.

In [23]:
merge3['housing'].value_counts()


yes        19433
no         16769
unknown      882
Name: housing, dtype: int64

### 19433 customers have housing loan.

In [24]:
merge3['loan'].value_counts()

no         30561
yes         5641
unknown      882
Name: loan, dtype: int64

### From the data we can infer that personal loan is availed by less no. of customers.

In [25]:
merge3['contact'].value_counts()

cellular     23522
telephone    13562
Name: contact, dtype: int64

### According to the data we can infer that most of the customers are contacted through cell phone rather than telephone. It means that marketting campaign can be well advertised through cellular phone rather than telephone.

In [26]:
merge3['month'].value_counts()

may    12420
jul     6430
aug     5562
jun     4786
nov     3705
apr     2388
oct      630
sep      510
mar      487
dec      166
Name: month, dtype: int64

### In the month of May, frequency of the bank marketting campaign was high and in the December month least marketting campaign was done.

In [27]:
merge3['day_of_week'].value_counts()

thu    7778
mon    7671
wed    7341
tue    7262
fri    7032
Name: day_of_week, dtype: int64

### Most of the marketting campaign was done on thursday, 7778 and less no. of marketting campaign was done on friday i.e 7032.

## Mean


In [55]:
merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].mean()

age                 40.042714
duration           258.237946
campaign             2.569545
previous             0.172986
emp.var.rate         0.082669
cons.price.idx      93.576076
cons.conf.idx      -40.505183
euribor3m            3.621668
nr.employed       5167.058664
dtype: float64

## Standard Deviation 


In [56]:
merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].std()

age                10.432965
duration          258.730909
campaign            2.770611
previous            0.495681
emp.var.rate        1.568997
cons.price.idx      0.578493
cons.conf.idx       4.622045
euribor3m           1.733972
nr.employed        72.196605
dtype: float64

## Variance


In [57]:
merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].var()

age                 108.846758
duration          66941.683369
campaign              7.676284
previous              0.245700
emp.var.rate          2.461752
cons.price.idx        0.334654
cons.conf.idx        21.363301
euribor3m             3.006659
nr.employed        5212.349718
dtype: float64

## Quantile


In [58]:
quantile1 = merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].quantile(q = 0.25)
print(quantile1)

age                 32.000
duration           102.000
campaign             1.000
previous             0.000
emp.var.rate        -1.800
cons.price.idx      93.075
cons.conf.idx      -42.700
euribor3m            1.344
nr.employed       5099.100
Name: 0.25, dtype: float64


In [59]:
quantile2 = merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].quantile(q = 0.50)
print(quantile2)

age                 38.000
duration           180.000
campaign             2.000
previous             0.000
emp.var.rate         1.100
cons.price.idx      93.749
cons.conf.idx      -41.800
euribor3m            4.857
nr.employed       5191.000
Name: 0.5, dtype: float64


In [60]:
quantile3 = merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].quantile(q = 0.75)
print(quantile3)

age                 47.000
duration           319.250
campaign             3.000
previous             0.000
emp.var.rate         1.400
cons.price.idx      93.994
cons.conf.idx      -36.400
euribor3m            4.961
nr.employed       5228.100
Name: 0.75, dtype: float64


## Skewness


In [61]:
merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].skew()

age               0.791149
duration          3.227855
campaign          4.743603
previous          3.857052
emp.var.rate     -0.722887
cons.price.idx   -0.229499
cons.conf.idx     0.297522
euribor3m        -0.709765
nr.employed      -1.044873
dtype: float64

#### Age is moderately skewed  that indicate asymmetry in the distribution. 
#### Duration, Campaign and previous are also highly skewed..
#### Employment variance rate, Consumer price index, euribor 3 month rate are negatively skewed indicating asymmetry in the distribution.



## Kurtosis

In [62]:
merge3[['age', 'duration', 'campaign', 'previous', 'emp.var.rate', 'cons.price.idx', 'cons.conf.idx', 'euribor3m', 'nr.employed' ]].kurtosis()

age                0.819752
duration          19.931726
campaign          36.838540
previous          20.494177
emp.var.rate      -1.064707
cons.price.idx    -0.829896
cons.conf.idx     -0.370663
euribor3m         -1.405760
nr.employed       -0.000882
dtype: float64

#### Kurtosis of (duration, campaign, pdays, previous) > 3; 
#### Therefore, it is Leptokurtic.

#### Kurtosis for (emp.var.rate, cons.price.id, nr.employed,cons.conf.idx, euribor3m) < 3;
#### Therefore, it is Platykurtic.


# Analysis of the response data with other attributes:

In [37]:
# lets get some more understanding of data and relationships
print('check 1: ', merge3.groupby(['loan','y']).size())
print('\n\ncheck 2: ', merge3.groupby(['job','y']).size())
print('\n\ncheck 3: ', merge3.groupby(['marital','y']).size())
print('\n\ncheck 4: ', merge3.groupby(['education','y']).size())
print('\n\ncheck 5: ', merge3.groupby(['poutcome','y']).size())
print('\n\ncheck 6: ', merge3.groupby(['month','y']).size())
print('\n\ncheck 7: ', merge3.groupby(['y', 'default']).size())



check 1:  loan     y  
no       no     27062
         yes     3499
unknown  no       791
         yes       91
yes      no      5023
         yes      618
dtype: int64


check 2:  job            y  
admin.         no     8174
               yes    1246
blue-collar    no     7738
               yes     576
entrepreneur   no     1198
               yes     111
housemaid      no      853
               yes      93
management     no     2345
               yes     292
retired        no     1146
               yes     395
self-employed  no     1145
               yes     131
services       no     3249
               yes     298
student        no      527
               yes     240
technician     no     5439
               yes     657
unemployed     no      791
               yes     134
unknown        no      271
               yes      35
dtype: int64


check 3:  marital   y  
divorced  no      3699
          yes      427
married   no     20187
          yes     2292
single    no      8930

##### In the above analysis we used groupby() to see the relationship between response column and other attribiutes.
##### 27062 customers didn't have personal loan and they didnt subsribed for term deposit.
##### 3499 of the customer doesn't have personal loan but they subscribed for term deposit.
##### 5023 customers have personal loan but they didn't subscribed for term deposit.
##### 618 customers have personal loan and they subscribed term deposit.
##### According to the above data, customers who subscribed for the term deposit are mostly from administration:1246 and the least subscribed are unemployed: 134.
##### Married customer are affecting the term deposit alot. Married customer who availed the term deposit are 2292 and non availed married customers are: 20187
##### May is the only month where there are high rate of customers who subscribed for the term deosit: 813







## Statistical Analysis on numeric columns


#### In the dataset we have both categorical and numerical columns. Let's look at the values of numeric columns first.

In [47]:
merge3[['pdays', 'campaign', 'previous']].describe()


pdays      campaign      previous
count  37084.000000  37084.000000  37084.000000
mean     962.530849      2.569545      0.172986
std      186.773063      2.770611      0.495681
min        0.000000      1.000000      0.000000
25%      999.000000      1.000000      0.000000
50%      999.000000      2.000000      0.000000
75%      999.000000      3.000000      0.000000
max      999.000000     56.000000      7.000000

##### According to the above observation we infer that minimum no. of days that passed by after client was last contacted from     previous campaign is 0.
##### The minimum no. of contacts performed during the campaign and for this the client last contacted was only 1.
##### Maximum times the no.of contacts performed during the campaign and for this the client last contacted was 56.
##### The no. of minimum contacts performed before the campaign and for this client was contacted is 0.
##### The no. of maximum contacts performed before the campaign and for this client was contacted is 7.


# Analysis of the response column


##### It is very important to look at the response column, which holds the information, which we are going to predict. In our case we should look at 'y' column and compare its values to other columns. First of all we should look at the number of 'yes' and 'no' values in the response column 'y'


##### Now let's look how numerical columns affect term deposit subscription.



In [48]:
#age and target column y

a_df = pd.DataFrame()
a_df['age_yes'] = (merge3[merge3['y'] == 'yes'][['y','age']].describe())['age']
a_df['age_no'] = (merge3[merge3['y'] == 'no'][['y','age']].describe())['age']

a_df

age_yes       age_no
count  4208.000000  32876.00000
mean     40.962928     39.92493
std      13.865754      9.90220
min      17.000000     17.00000
25%      31.000000     32.00000
50%      37.000000     38.00000
75%      50.000000     47.00000
max      98.000000     95.00000

##### The minimum age of the customer who availed the term deposit is of 17 years.
##### The average age of the customer's who availed the term deposit/bank scheme is of 40.96 yeras.
##### The maximum age of the customer who availed the term deposit is of 98 years.
##### The maximum age of the customer who didn't availed the term deposit is of 95 years.
##### The average age of the customer's who didn't availed the term deposit/bank scheme is of 39.92 yeras.



In [54]:
#number of contacts performed during this campaign ('campaign') and target column y
c_df = pd.DataFrame()
c_df['campaign_yes'] = (merge3[merge3['y'] == 'yes'][['y','campaign']].describe())['campaign']
c_df['campaign_no'] = (merge3[merge3['y'] == 'no'][['y','campaign']].describe())['campaign']

c_df

campaign_yes   campaign_no
count   4208.000000  32876.000000
mean       2.048004      2.636300
std        1.661395      2.875126
min        1.000000      1.000000
25%        1.000000      1.000000
50%        2.000000      2.000000
75%        2.000000      3.000000
max       23.000000     56.000000

####  Average no. of contacts performed during the campaign who agreed to the term deposit are 2.04 and the one who didn't agreed to the term deposit/bank marketting campagin are 2.64
#### Minimum no. of contacts performed during the campaign who agreed to the term deposit is 1 and who didn't agreed ia also 1.
#### Maximum no. of contacts performed during the campaign who agreed to the term deposit are 23 and the minimum no. of contacts performed during the campaign who agreed to the term deposit are 5


In [53]:
#number of contacts performed during previous campaign ('previous') and target column y
p_df = pd.DataFrame()
p_df['previous_yes'] = (merge3[merge3['y'] == 'yes'][['y','previous']].describe())['previous']
p_df['previous_no'] = (merge3[merge3['y'] == 'no'][['y','previous']].describe())['previous']

p_df

previous_yes   previous_no
count   4208.000000  32876.000000
mean       0.493584      0.131950
std        0.861150      0.409158
min        0.000000      0.000000
25%        0.000000      0.000000
50%        0.000000      0.000000
75%        1.000000      0.000000
max        6.000000      7.000000

#### Average no. of contacts performed before the campaign is 0.49 who agreed for the term deposit and the one who didn't agreed are 0.13.
#### Minimum no. of contacts performed before the campaign is 0 who agreed for the term deposit and the one who didn't agreed were also 0.
####  Maximum no. of contacts performed before the campaign are 6 who agreed for the term deposit and the one who didn't agreed are 7

#  Task 1.3 Statistical Analysis using Python
###  Q2: Performing relevant hypothesis testing

### a)_To check whether mode of Contact has any correlation with the response of the customer using the Chi Square Test._
#### Null Hypothesis: Response of the customer does not depend on contact
#### Alternate Hypothesis: Response of the customer depends on contact

In [7]:
import pandas as pd
import numpy as np
from pandas import Series, DataFrame
import scipy.stats as st

In [8]:
#Loading the customer data csv files"

df5=pd.read_csv("C:/Users/anwpatnaik/Downloads/Capstone Project1_Marketing Data/Capstone Project1_Marketing Data/Marketing_Campaign_P1/Customer_campaign_details_p1.csv")
df6=pd.read_csv("C:/Users/anwpatnaik/Downloads/Capstone Project1_Marketing Data/Capstone Project1_Marketing Data/Marketing_Campaign_P1/Customer_response_data_p1.csv")
df7=pd.read_csv("C:/Users/anwpatnaik/Downloads/Capstone Project1_Marketing Data/Capstone Project1_Marketing Data/Marketing_Campaign_P1/Customer_social_economic_data_p1.csv")
Customer_and_bank_details = pd.read_csv("C:/Users/anwpatnaik/Downloads/Capstone Project1_Marketing Data/Capstone Project1_Marketing Data/Marketing_Campaign_P1/Customer_and_bank details_p1.csv")

In [9]:
#Computing cross tabulation of 2 columns- Response and Contact

chi=pd.crosstab(df6.y,df5.contact)

In [10]:
chi

contact  cellular  telephone
y                           
no          20030      12846
yes          3492        716

In [11]:
#Allocating the values of dataframe Chi into an array

chi2= np.array([chi.iloc[0][:2].values, chi.iloc[1][:2].values])

In [12]:
chi2

array([[20030, 12846],
       [ 3492,   716]], dtype=int64)

In [13]:
st.chi2_contingency(chi2)[0:2]

(781.5964619543073, 5.412272241321877e-172)

#### As  P Value is very less(Null), we are accepting the Alternate Hypothesis.
#### Which means that the Reponse of the customer depends on the mode of Contact(Cellular/Telephone)

### b) _To check whether Loan has any correlation with the response of the customer using the Chi Square Test._
#### Null Hypothesis: Response of the customer does not depend on Loan
#### Alternate Hypothesis: Response of the customer depends on Loan

In [15]:
#Computing cross tabulation of 2 columns- Response and loan
   
chi3=pd.crosstab(Customer_and_bank_details["loan"],df6.y)

In [16]:
chi3

y           no   yes
loan                
no       27062  3499
unknown    791    91
yes       5023   618

In [17]:
#Allocating the values of dataframe Chi3 into an array

chi2= np.array([chi3.iloc[0][:2].values, chi3.iloc[1][:2].values,chi3.iloc[2][:2].values])

In [18]:
chi2

array([[27062,  3499],
       [  791,    91],
       [ 5023,   618]], dtype=int64)

In [19]:
st.chi2_contingency(chi2)[0:3]

(2.1063167111252143, 0.34883426478989643, 2)

#### As P Value is higher than Alpha(0.05), hence accepting the Null Hypothesis
#### Which means that the response of the customer does not depend on Loan

### c) _To check whether the age is dependent on job or not using ANOVA Test_
  
#### Null Hypothesis: Mean age is independent of Job.
#### Alternate Hypothesis: Mean age is dependent on Job.

In [20]:
services = Customer_and_bank_details[Customer_and_bank_details['job'] == 'services']['age']
admin = Customer_and_bank_details[Customer_and_bank_details['job'] == 'admin.']['age']
blue_Collar = Customer_and_bank_details[Customer_and_bank_details['job'] == 'blue-collar']['age']
technician = Customer_and_bank_details[Customer_and_bank_details['job'] == 'technician']['age']
housemaid = Customer_and_bank_details[Customer_and_bank_details['job'] == 'housemaid']['age']
retired = Customer_and_bank_details[Customer_and_bank_details['job'] == 'retired']['age']
management = Customer_and_bank_details[Customer_and_bank_details['job'] == 'management']['age']
self_employed = Customer_and_bank_details[Customer_and_bank_details['job'] == 'self-employed']['age']
student = Customer_and_bank_details[Customer_and_bank_details['job'] == 'student']['age']
   

In [21]:
st.jarque_bera(services)

Jarque_beraResult(statistic=176.91423648001214, pvalue=0.0)

#### As p value is obtained using JB Test is less than 0.05 hence

#### Reject the null hypothesis

#### Hence sample does not follow Normal Distribution thus we can't apply parametric test